components/brave_shields/core/browser/ad_block_filter_list_catalog_provider.cc

```diff
@@ -11,13 +11,30 @@
 #include "base/files/file_path.h"
 #include "base/task/thread_pool.h"
 #include "brave/components/brave_shields/core/browser/ad_block_component_installer.h"
+#include "brave/components/brave_shields/resources/grit/adblock_bundle.h"
+#include "components/component_updater/component_updater_service.h"
+#include "ui/base/resource/resource_bundle.h"
 
 constexpr char kListCatalogFile[] = "list_catalog.json";
 
+namespace {
+
+std::string LoadBundledResource() {
+  return ui::ResourceBundle::GetSharedInstance().LoadDataResourceString(
+      IDR_BRAVE_ADBLOCK_LIST_CATALOG);
+}
+
+}  // namespace
+
 namespace brave_shields {
 
 AdBlockFilterListCatalogProvider::AdBlockFilterListCatalogProvider(
     component_updater::ComponentUpdateService* cus) {
+  base::SequencedTaskRunner::GetCurrentDefault()->PostTaskAndReplyWithResult(
```

Comment:
> base::SequencedTaskRunner::GetCurrentDefault()

What thread we use to parse this? UI?

Reply:
the thread it gets parsed on doesn't appear to change from the list loaded from the crx

Reply:
`base::SequencedTaskRunner::GetCurrentDefault()` referrers to UI here.
We should post to the thread pool or call it directly.

Reply:
`LoadBundledResource` does not parse the list, it just returns the string contents from resource bundle which is fine to do on the UI thread because there is no blocking IO. The rest of the process is the same as loading it from a file

Reply:
I'm not even sure why it's doing a PostTask here instead of just passing the string directly to `OnFilterListCatalogLoaded`. As a side note, it would be better to use a lambda here instead of `LoadBundledResource`